src/modules/transaction/hooks/useTransactionFee/useTransactionFee.js

```diff
@@ -1,69 +1,61 @@
-import { getParamsSchema } from './utils';
-import usePriorityFee from '../usePriorityFee';
+/* eslint-disable complexity */
 import { FEE_TYPES } from '../../constants';
 import { joinModuleAndCommand } from '../../utils';
-import { useMinimumFee } from '.';
-import { MODULE_COMMANDS_NAME_MAP } from '../../configuration/moduleCommand';
+import { useTransactionEstimateFees } from '../queries';
 
 // eslint-disable-next-line max-statements
-export const useTransactionFee = ({
-  isFormValid,
-  transactionJSON,
-  senderAddress,
-  selectedPriority = [],
-  extraCommandFee = 0,
-}) => {
-  const {
-    moduleCommandSchemas,
-    isLoading: isSchemaLoading,
-    isFetched: isSchemaFetched,
-  } = useCommandSchema();
-  const paramsSchema = getParamsSchema(transactionJSON, moduleCommandSchemas);
+export const useTransactionFee = ({ isFormValid, transactionJSON, extraCommandFee = 0 }) => {
+  let accountInitializationFee = {};
+  const { signatures, ...transactionToSendJSON } = transactionJSON;
 
-  const {
-    result: minimumFee,
-    isLoading: isLoadingByteFee,
-    isFetched: isFetchedByteFee,
-  } = useMinimumFee({
-    senderAddress,
-    isFormValid,
-    transactionJSON,
-    extraCommandFee,
-  });
+  transactionToSendJSON.fee = transactionToSendJSON.fee.toString();
 
-  const priorityFee = usePriorityFee({
-    selectedPriority,
-    transactionJSON,
-    paramsSchema,
-    isEnabled: !!paramsSchema && isFormValid,
+  const transactionFee = useTransactionEstimateFees({
+    config: { data: { transaction: transactionToSendJSON } },
+    options: { enabled: isFormValid && !!transactionJSON },
   });
+
+  const { data: { transaction } = {}, meta = {} } = transactionFee?.data || {};
+  const moduleCommand = joinModuleAndCommand(transactionJSON);
+  const initializationFee =
+    meta.breakdown?.fee?.minimum?.additionalFees?.userAccountInitializationFee || 0;
+  const minimumFee = BigInt(transaction?.fee?.minimum || 0);
+  const messageFee = BigInt(transaction?.params?.messageFee?.amount || 0);
+
+  if (initializationFee) {
+    accountInitializationFee = {
+      value: BigInt(initializationFee),
+      type: 'Account Initialization',
+    };
+  }
+
+  const priorityFee = {
+    type: FEE_TYPES.PRIORITY_FEE,
+    value: 0,
+  };
   const bytesFee = {
-    value: BigInt(minimumFee.value) - BigInt(extraCommandFee),
+    value: BigInt(meta.breakdown?.fee?.minimum?.byteFee || 0),
     type: FEE_TYPES.BYTES_FEE,
   };
-  const components = [bytesFee, priorityFee].filter((item) => item.value > 0);
-  const moduleCommand = joinModuleAndCommand(transactionJSON);
-  let txComponentType = '';
+  let components = [bytesFee, accountInitializationFee, priorityFee].filter(
+    (item) => item.value > 0
+  );
+
   if (
-    moduleCommand === MODULE_COMMANDS_NAME_MAP.transfer ||
-    moduleCommand === MODULE_COMMANDS_NAME_MAP.transferCrossChain
+    moduleCommand !== 'token:transfer' &&
+    moduleCommand !== 'token:transferCrossChain' &&
```

Comment:
use the map MODULE_COMMANDS_NAME_MAP.transfer , MODULE_COMMANDS_NAME_MAP.transferCrossChain?